coremltools/converters/mil/mil/ops/defs/iOS15/scatter_gather.py

```diff
@@ -486,26 +486,35 @@ class scatter_nd(Operation):
     Scatter ``updates`` to ``data`` at locations ``indices``.
 
     The ``indices`` is a K-dim tensor, where ``indices[i_0,...,i_{K-2}]`` defines a
-    slice of ``data``, ``K = rank(indices)``, and ``data[indices[i_0, ..., i_{K-2}]]``
-    has rank ``rank(data) - indices.shape[-1]``.
+    slice of ``data``, ``K = rank(indices)``, and ``data[i_0, ..., i_{K-2}]``
+    has rank ``rank(data) - indices.shape[-1]``. Concretely, this means the index is
+    stored in the last dim of ``indices``, e.g. take a ``K == 2`` example
+
+    .. math::
+       indices = [[0, 1],
+                  [0, 2]]
+
+    where ``indices[0]`` / ``[0, 1]`` and ``indices[1]`` / ``[0, 2]`` are
+    two indices that get applied to ``data``
 
     * Example: ``mode == update``: The ``output`` is set to ``data`` initially, and
       the op updates ``output`` as follows:
 
     .. math::
-       output[indices[i_0, ..., i_{K-2}]]= updates[indices[i_0, ..., i_{K-2}]]
+       output[indices[i_0, ..., i_{K-2}]]= updates[i_0, ..., i_{K-2}]
 
     * Example: ``mode == add``. The update rule is:
 
     .. math::
-       output[indices[i_0, ..., i_{K-2}]] += updates[indices[i_0, ..., i_{K-2}]]
+       output[indices[i_0, ..., i_{K-2}]] += updates[i_0, ..., i_{K-2}]
 
     Parameters
     ----------
     data: tensor<\*D, T> (Required)
-    indices: tensor<\*K, i32> (Required)
-    updates: tensor<\*K, T> (Required)
-        * Must be the shape as ``K[:-1]+data.shape[K[-1]:]``.
+    indices: tensor<\*E, i32> (Required)
```

Comment:
This should still be `K`?
As we have `K = rank(indices)` in the doc string?

Reply:
Oh so this was intentional?

Concretely, we were intentionally overloading symbol `K` to both be `data.shape` and `data.rank`?

Reply:
I think we are keeping consistency, yes.

Reply:
Oh okay you are right, overloading the shape and rank might not be a good idea either haha.
At least, we should distinguish the shape of indices from updates?
Keep both of them `E` might be confusing.

Reply:
Wow great catch! It was another error in original doc that fooled me: `indices.shape` and `update.shape` were both `*K`

Revised `update.shape` to be `*F`